transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/StepRunTransformations.scala

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.monster.encode.transformation
+
+import org.broadinstitute.monster.encode.jadeschema.table.StepRun
+import upack.Msg
+
+/** Transformation logic for ENCODE analysis step run objects. */
+object StepRunTransformations {
+  import org.broadinstitute.monster.common.msg.MsgOps
+
+  /** Transform a raw ENCODE analysis step run into our preferred schema. */
+  def transformStepRun(
+    rawStepRun: Msg,
+    rawStepVersion: Msg,
+    rawStep: Msg,
+    rawGeneratedFiles: Iterable[Msg],
+    fileIdToTypeMap: Map[String, FileType]
+  ): StepRun = {
+
+    // get pipeline run id
+    val stepRunId = CommonTransformations.readId(rawStepRun)
+    val pipelineId = PipelineRunTransformations.getPipelineId(rawStep)
+    val experimentId = PipelineRunTransformations.getExperimentId(rawGeneratedFiles, stepRunId)
+    val pipelineRunId =
+      if (pipelineId.isEmpty || experimentId.isEmpty) None
+      else Some(PipelineRunTransformations.getPipelineRunId(pipelineId.head, experimentId.head))
```

Comment:
Instead of the `if`-`else`, more idiomatic Scala would be to `flatMap`:
```scala
val pipelineRunId = for {
  pipelineId <- PipelineRunTransformations.getPipelineId(rawStep)
  experimentId <- PipelineRunTransformations.getExperimentId(rawGeneratedFiles, stepRunId)
} yield {
  PipelineRunTransformations.getPipelineRunId(pipelineId, experimentId)
}
```

Reply:
Thanks! I hadn't used yield in Scala before, but it seems really helpful. I put that pattern in a couple of other places too... let me know if I'm going overboard 😁